File: backend/app/models/analysis.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class QAAnswer(BaseModel):
    question: str
    yourAnswer: str
    score: int
    category: str
    aiSuggestion: Optional[str] = None
    improvedAnswer: Optional[str] = None

class AnalysisResult(BaseModel):
    interviewId: str
    duration: str
    rounds: int
    score: float
    passRate: int
    strengths: List[dict]
    weaknesses: List[dict]
    qaList: List[QAAnswer]
    suggestions: List[dict]
    createdAt: datetime
    updatedAt: datetime

File: src/types/analysis.ts
export interface QAItem {
  id: number;
  question: string;
  questioner?: string;
  questionTime?: string;
  answer: string;
  answerer?: string;
  answerTime?: string;
  notes?: string;
  score?: number;
  category?: string;
}

export interface StrengthItem {
  title: string;
  desc: string;
}

export interface SuggestionItem {
  title: string;
  desc: string;
  priority: '高' | '中' | '低';
  actions: string[];
}

export interface AnalysisData {
  duration: string;
  rounds: number;
  score: number;
  passRate: number;
  strengths: StrengthItem[];
  weaknesses: StrengthItem[];
  qaList: QAItem[];
  suggestions: SuggestionItem[];
  quickSummary?: string;
  generatedAt?: string;
}
